src/coreComponents/constitutive/fluid/multifluid/PVTDriver.cpp

```diff
@@ -186,22 +220,39 @@ bool PVTDriver::execute( real64 const GEOS_UNUSED_PARAM( time_n ),
   return false;
 }
 
-
-
 void PVTDriver::outputResults()
 {
   // TODO: improve file path output to grab command line -o directory
   //       for the moment, we just use the specified m_outputFile directly
 
   FILE * fp = fopen( m_outputFile.c_str(), "w" );
 
-  fprintf( fp, "# column 1 = time\n" );
-  fprintf( fp, "# column 2 = pressure\n" );
-  fprintf( fp, "# column 3 = temperature\n" );
-  fprintf( fp, "# column 4 = density\n" );
-  fprintf( fp, "# columns %d-%d = phase fractions\n", 5, 4+m_numPhases );
-  fprintf( fp, "# columns %d-%d = phase densities\n", 5+m_numPhases, 4+2*m_numPhases );
-  fprintf( fp, "# columns %d-%d = phase viscosities\n", 5+2*m_numPhases, 4+3*m_numPhases );
+  integer columnIndex = 0;
+  fprintf( fp, "# column %d = time\n", ++columnIndex );
+  fprintf( fp, "# column %d = pressure\n", ++columnIndex );
+  fprintf( fp, "# column %d = temperature\n", ++columnIndex );
+  fprintf( fp, "# column %d = density\n", ++columnIndex );
+  if( m_outputCompressibility != 0 )
+  {
+    fprintf( fp, "# column %d = total compressibility\n", ++columnIndex );
+  }
+  fprintf( fp, "# columns %d-%d = phase fractions\n", columnIndex+1, columnIndex + m_numPhases );
+  columnIndex += m_numPhases;
+  fprintf( fp, "# columns %d-%d = phase densities\n", columnIndex+1, columnIndex + m_numPhases );
+  columnIndex += m_numPhases;
+  fprintf( fp, "# columns %d-%d = phase viscosities\n", columnIndex+1, columnIndex + m_numPhases );
+  columnIndex += m_numPhases;
+
+  if( m_outputPhaseComposition != 0 )
+  {
+    auto const phaseNames = getFluid().phaseNames();
+    for( integer ip = 0; ip < m_numPhases; ++ip )
+    {
+      fprintf( fp, "# columns %d-%d = %s phase fractions\n", columnIndex+1, columnIndex + m_numComponents,
```

Comment:
may be convenient to repeat the component name order here to make the column meaning clear: `water phase component fractions [H20, CO2, CH4]`